### PR TITLE
Basic homepage

### LINKED ISSUE #149: implementatie van home page
home page van de website (gelijkend met de mock-up)
- definiëren van vaste kleuren
- links naar hun pagina's facebook, instagram, hun eigen website, ...
- nog geen afbeeldingen 
- hun logo

BASE_URL gebruiken (cfr. Discord)

### PR DESCRIPTION
**Pull request by Robindr145**
## 🎯 MAIN GOAL
- [ ] this PR implements the home page 

**Missing tests:**
* clarify if certain edge-cases still need testing

## 📝 DOCUMENTATION
- [x] there is enough documentation written for this code
- [x] external documentation (README, etc.) has been changed : default README.md in the frontend folder has been removed

**Missing documentation:**
* clarify where in the code extra documentation is required

## 🔍 HOW TO TEST
1. type `npm i` in the root
2. start the localhost with `npm run start:dev`
3. look if the hompage looks good + the favicon (URL) + the name (Viernulvier-Archive)

## ⚠️ REMAINING ISSUES
- [ ] there are still remaining bugs in the code
* explain where the bugs are located
- [ ] this code requires changes elsewhere (changed database structure, API-endpoint, etc.)
* clarify what needs to be changed

## ✅ CLOSED ISSUES
- #149

### CODE REVIEW

**Comment by MrLegitFTW** [comment]
Could you pull the latest changes from develop? That way the preview should deploy and we can check it out :-)

**Comment by Alikznollet** [comment]
npm error The "from" argument must be of type string. Received undefined 

Action failed. A type wrong somewhere?

**Comment by Robindr145** [comment]
> Alikznollet

@nuxtjs/i18n, tailwindcss, lucide-vue-next are in root and /frontend package.json. I'm going to fix this

**Comment by ddndauw** [comment]
Looks good to me for the moment! The home page already looks very professional, it also provides a good base to start with, good job!

Some (nitpicky) details:
- I don't know if "Selection" above "Archive Highlights" is needed? (But that also might just be my personal taste.)
selection

- Right now when you switch language, the title jumps a bit. I don't know if it's really possible to "fix" it at a certain position, but maybe you can add a bit more space so "Gent" can go on the same line as "Decennia van kunst, dans, muziek en theater in het hart van".
aaa

Again, the page looks very good already !

**Comment by jillvanham** [comment]
Everything looks great! Seems like a good start for our home page! I have a few critical points to think about, but all just details. 

- Right now, everything (buttons, production 'cards'...) is a sharp rectangle instead of being rounded like in the mock-up. Both are great but maybe we should think about what border-radius we want to use for the entire site, so everything follows the same convention. I really liked the look of the Figma mock-up (with the rounded borders), but I don't hate this either. 

- Maybe we should discuss in a group setting which fonts we want to use for the entire site. I see you're already using a global style sheet, which is perfect. That way future changes won't be a big problem. I actually like the font, so this is not criticism, just curious to check out different options. 

- I don't know if I'm a fan of the 'discover more' part. It doesn't look bad, but at first sight it just looks like some photos (in a carousel) , rather than links to new pages. Adding some padding between the cards or slightly reducing their width might help them look more like clickable elements.

- Like Diana mentioned earlier, I don't really get why we would put 'selection' above the 'Archive Highlights'. I think we can just leave that out. 

- The 'About Us' text has low readability. We might need to increase the size or increase the color contrast to make it more visible (but I do like the font).

**Comment by Nowak-Dominik** [comment]
@Robindr145 Here is my review of the visual aspects of Home page.

I find some of these examples not visible enough. Maybe change the font color a little bit so it is slightly more visible.
Examples:

image
image
image

Another thing that I noticed is about that part:

image

This is very nicely done but that section stretches so it takes the whole width which is fine on like normal sized screens but for example I am looking at this on a super wide screen. And then it is more noticible that this is the only section that does not focus it's content in the middle part of the screen like other do.

Here is what I mean:

image

image

Versus this one that is different:

image

As you can see the clickable stuff and the things you can read are spreaded as wel, so the main focus point of that section is not in the middel like in the other ones.

So I would like you to try to find a way to fix this in a way you think looks good. Maybe you can let the pictures stretch but keep the text and buttons in a fixed place like other sections do.

All other stuff visually looks good to me and you did a great job.

**Comment by Nowak-Dominik** [comment]
@Robindr145 I also just noticed that the icon you made is not yet used:

image

image

File: frontend/app/pages/index.vue [hunk 1]
```diff
@@ -1,16 +1,368 @@
...
-const config = useRuntimeConfig();
+<script setup lang="ts">
+import { ArrowRight, Calendar, FileText, ImageIcon } from 'lucide-vue-next'
 
-const { error } = await useFetch("/", { baseURL: config.public.apiBase });
-const connected = !error.value;
+const { t, locale, setLocale } = useI18n()
+
+
+// mock data, replace with actual data !!!!!
```
Comment (Nowak-Dominik) [inline frontend/app/pages/index.vue]: Change this to a TODO comment so it's more visible especially becouse it is important
```suggestion
// TODO: mock data, replace with actual data !!!!!
```

File: frontend/app/assets/css/tailwind.css [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  }
+
+  input {
+    font-size: var(--text-base);
+    font-weight: var(--font-weight-normal);
+    line-height: 1.5;
+  }
+}
+
+.dropcap::first-letter {
```
Comment (Nowak-Dominik) [inline frontend/app/assets/css/tailwind.css:117]: Maybe these comments are temporary, but write them in English.

File: frontend/app/pages/index.vue [hunk 3]
```diff
@@ -1,16 +1,368 @@
...
+    <!-- ═══════════════════════════════════════ HERO ═══════════════════════════════════════ -->
+    <section class="relative h-[85vh] min-h-[560px] overflow-hidden bg-black">
+      <img
+        src="https://images.unsplash.com/photo-1765278624799-9c90305b0b7e?w=1400&q=80"
+        alt="VIERNULVIER performance"
+        class="absolute inset-0 w-full h-full object-cover opacity-50 hero-img"
+      />
+      <div class="absolute inset-0 bg-gradient-to-t from-black via-black/40 to-transparent" />
+
+      <!-- Language toggle (tijdelijk hier, later naar header) -->
```
Comment (Nowak-Dominik) [inline frontend/app/pages/index.vue]: Same here, write comments in English.

File: frontend/app/pages/index.vue [hunk 4]
```diff
@@ -1,16 +1,368 @@
...
+        </p>
+        <h2 class="text-3xl md:text-4xl font-display font-black">
+          {{ t('discover.title') }}
+        </h2>
+      </div>
+
+      <!-- Full-bleed image cards -->
+      <div class="grid md:grid-cols-2" style="height: 520px;">
+
+        <!-- Verhalen card -->
```
Comment (Nowak-Dominik) [inline frontend/app/pages/index.vue]: Maybe you should define these as a component becouse Blogs/verhalen card and Uploads/affiches card is basicly the same just with diffrent text, pictures and link.

File: frontend/app/pages/index.vue [hunk 5]
```diff
@@ -1,16 +1,368 @@
...
+              {{ t('discover.posters.cta') }}
+              <ArrowRight class="w-4 h-4 group-hover:translate-x-1 transition-transform" />
+            </div>
+          </div>
+        </NuxtLink>
+
+      </div>
+    </section>
+
+    <!-- ════════════════════════════════════ FOOTER ════════════════════════════════════ -->
```
Comment (Nowak-Dominik) [inline frontend/app/pages/index.vue]: Maybe define a Footer as a separate component as wel becouse technicly on every page when the user goes to the bottom of the screen he/she should see the same footer there. It's like the header but at the bottom.
If you do this we will be able to just use an instantiation of the footer everywhere becouse it has a constant layout on every page.

File: frontend/app/pages/index.vue [hunk 6]
```diff
@@ -1,16 +1,368 @@
...
+
+<style scoped>
+.font-display {
+  font-family: 'Georgia', 'Times New Roman', serif;
+}
+.font-mono {
+  font-family: 'Courier New', Courier, monospace;
+}
+
+/* VIERNULVIER — groot en bold */
```
Comment (Nowak-Dominik) [inline frontend/app/pages/index.vue]: Again, English please

File: frontend/app/pages/index.vue [hunk 7]
```diff
@@ -1,16 +1,368 @@
...
+.hero-title {
+  font-size: clamp(3.5rem, 10vw, 8rem);
+}
+
+/* Archief — iets kleiner, lichtgewicht, sluit aan op de mock-up */
```
Comment (Nowak-Dominik) [inline frontend/app/pages/index.vue]: Again, English please

File: frontend/app/pages/index.vue [hunk 8]
```diff
@@ -1,16 +1,368 @@
...
+        <NuxtLink to="/archive">
+          <button class="group inline-flex items-center gap-3 bg-white text-black px-7 py-3.5 font-bold text-sm tracking-wide hover:bg-white/90 transition-colors">
+            {{ t('hero.cta') }}
+            <ArrowRight class="w-4 h-4 group-hover:translate-x-1 transition-transform" />
+          </button>
+        </NuxtLink>
+      </div>
+    </section>
+
+    <!-- ══════════════════════════════ ARCHIVE HIGHLIGHTS ══════════════════════════════ -->
```
Comment (Nowak-Dominik) [inline frontend/app/pages/index.vue]: Here you define Grid views of production items which should be defined separetly in as a component.
There is even an issue for this: #169

File: frontend/app/assets/css/tailwind.css [hunk 9]
```diff
@@ -0,0 +1,192 @@
+@import "tailwindcss";
+
+@custom-variant dark (&:is(.dark *));
```
Comment (Nowak-Dominik) [inline frontend/app/assets/css/tailwind.css:22]: Change some of these parameters so you adress the things me and others said in the visual review of this pr.

**Review by Nowak-Dominik — CHANGES_REQUESTED** [review]
@Robindr145 
I reviewd your pr code wise. Check the comments I left for you in the code.

File: frontend/app/components/Footer.vue [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+            </a>
+          </div>
+
+          <a
+            href="https://www.viernulvier.gent/nl/newsletter-inschrijven-q1lr"
+            target="_blank"
+            rel="noopener noreferrer"
+            class="inline-flex items-center gap-2 px-5 py-2 border border-white dark:border-black text-[0.7rem] font-bold tracking-widest uppercase hover:bg-white hover:text-black dark:hover:bg-black dark:hover:text-white transition-colors whitespace-nowrap"
+          >
+            {{ t('footer.newsletter') }} →
```
Comment (Alikznollet) [inline frontend/app/components/Footer.vue]: Maybe look into removing this because we don't really need/have a footer in the archive.

File: frontend/app/pages/index.vue
Comment (Alikznollet) [inline frontend/app/pages/index.vue:1]: Maybe try to split this page up into different components, if that's not already in the works. File might be getting a little too big.

**Review by Alikznollet — COMMENTED** [review]
Some small things I noticed while skimming through the preview and code. We're not using a footer in our archive and don't have the functionality for that so maybe you can remove it?

The other comment is optional but could be useful.

**Comment by Robindr145** [comment]
I think we should discuss about the fonts and colours we want in our website in the next meeting.

**Comment by MrLegitFTW** [comment]
> I think we should discuss about the fonts and colours we want in our website in the next meeting.

This seems like a good idea. Could we plan an interim meeting if possible through discord? That way this can be updated quicker.

File: .github/workflows/pr-preview.yml [hunk 11]
```diff
@@ -40,7 +40,7 @@ jobs:
       # STEP 2: Build & Test
       - name: Install and Test
         run: |
-          npm i
```
Comment (Nowak-Dominik) [inline .github/workflows/pr-preview.yml]: What is this change for? Could you explain shortly @Robindr145

File: frontend/app/components/home/Highlights.vue [hunk 12]
```diff
@@ -0,0 +1,136 @@
+<script setup lang="ts">
+import { ArrowRight, Calendar } from 'lucide-vue-next'
+
+const { t, locale } = useI18n()
+
+// TODO: replace with real API data once the database is connected
```
Comment (Nowak-Dominik) [inline frontend/app/components/home/Highlights.vue:33]: Technicly you already can acces some data but its fine that you have the hardcoded examples list for now becouse the GridViewItem for production needs to be defined anyway in a separate pr.

File: frontend/app/components/Footer.vue [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+                {{ t('footer.mainSite') }}
+              </a>
+            </li>
+            <li><NuxtLink to="/archive" class="hover:text-white/60 dark:hover:text-black/60 transition-colors">{{ t('footer.archive') }}</NuxtLink></li>
+            <li><NuxtLink to="/blogs" class="hover:text-white/60 dark:hover:text-black/60 transition-colors">{{ t('footer.stories') }}</NuxtLink></li>
+            <li><NuxtLink to="/uploads" class="hover:text-white/60 dark:hover:text-black/60 transition-colors">{{ t('footer.prints') }}</NuxtLink></li>
+          </ul>
+        </div>
+
+        <!-- Social media icon links -->
```
Comment (Nowak-Dominik) [inline frontend/app/components/Footer.vue]: Maybe you could define al of these social media logo's svg's in assets folder and then here you could just reference so the code is more clear.

File: .github/workflows/pr-preview.yml (hunk 11 above)
Comment (Robindr145) [reply .github/workflows/pr-preview.yml]: I think @MrLegitFTW did this, but I think with this change, the server starts with a clean npm install, this small change fixed the first cause of errors with the server tests.

File: frontend/app/pages/index.vue
Comment (Nowak-Dominik) [inline frontend/app/pages/index.vue:1]: Would it not be better that you define the home page in a separate file and then in index.vue we could have like the Header + the page that should be displayed + footer?

File: frontend/app/components/home/Discover.vue [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+            </h3>
+            <p class="text-sm text-white/70 leading-relaxed mb-6 max-w-sm">
+              {{ t('discover.stories.body') }}
+            </p>
+            <div class="inline-flex items-center gap-3 text-white font-mono text-xs font-bold uppercase tracking-widest border-b border-white/30 pb-1 w-fit group-hover:border-white transition-colors duration-300">
+              {{ t('discover.stories.cta') }}
+              <ArrowRight class="w-4 h-4 group-hover:translate-x-1 transition-transform" />
+            </div>
+          </div>
+        </NuxtLink>
```
Comment (Nowak-Dominik) [inline frontend/app/components/home/Discover.vue]: Make this as a component becouse now you have duplicate code for Stories and Posters

File: frontend/app/components/home/Discover.vue [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+      <h2 class="text-3xl md:text-4xl font-display font-black mb-12">
+        {{ t('discover.title') }}
+      </h2>
+
+      <!-- Two cards side by side, constrained to max-w-7xl like all other sections.
+           Gap makes them feel like distinct clickable elements rather than a single image strip. -->
+      <div class="grid md:grid-cols-2 gap-4 md:gap-6">
+
+        <!-- Stories & History card -->
+        <NuxtLink to="/blogs" class="group relative overflow-hidden block" style="height: 460px;">
```
Comment (Nowak-Dominik) [inline frontend/app/components/home/Discover.vue]: Use `routes.ts` instead of hardcoding the /blogs link.
You do this in multiple places so make sure that you use `routes.ts` if you want to reference a route.

**Review by Nowak-Dominik — CHANGES_REQUESTED** [review]
Check my comments I placed in the files.
Also a general remark: Add some additional documentation about what a specific file is about.
So in all of these files mention that this file implements ...
And if needed mention the important stuff that happends in that file so other people know directly what they are looking at.

**Comment by Nowak-Dominik** [comment]
Visuals wise I approve this pr. Good job, everything looks nice!

**Review by Alikznollet — COMMENTED** [review]
I'll wait to review this until the worker passes again.

**Comment by Robindr145** [comment]
@Alikznollet Visual wise, you can give your opinion with the website preview, I don't think I'm gonna change the UI aspects.

**Comment by Alikznollet** [comment]
> @Alikznollet Visual wise, you can give your opinion with the website preview, I don't think I'm gonna change the UI aspects.

Visual wise to me for now it looks fine!

File: frontend/app/pages/index.vue
Comment (Robindr145) [reply frontend/app/pages/index.vue:1]: I'm not sure if we want to display the footer on every page, will add that to the discussion points for the next meeting

File: frontend/app/components/Footer.vue (hunk 13 above)
Comment (Robindr145) [reply frontend/app/components/Footer.vue]: That's a good idea

File: frontend/app/components/home/Highlights.vue (hunk 12 above)
Comment (Robindr145) [reply frontend/app/components/home/Highlights.vue:33]: Yes, will fix it in the next PR

**Review by Alikznollet — CHANGES_REQUESTED** [review]
Before you merge this PR you should:

1. run `npm audit fix` and then `npm audit --omit=dev` to see whether all vulnerabilities are fixed for the production environment (I found some when on the branch).
2. Merge develop again because a LOT of changes happened.

This PR has been open for a long time so I think we would benefit off it just getting merged and tests being a seperate issue.

File: frontend/app/components/home/Hero.vue [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+
+    <!-- Hero content anchored to bottom -->
+    <div class="relative z-20 h-full flex flex-col justify-end px-6 md:px-12 lg:px-20 pb-16 max-w-7xl mx-auto w-full">
+      <!-- Title: VIERNULVIER bold + Archive/Archief light underneath -->
+      <div class="mb-8">
+        <h1 class="hero-title font-display font-black text-white uppercase leading-none tracking-tight">
+          VIERNULVIER
+        </h1>
+        <p class="hero-archief font-display font-light text-white/90 leading-none tracking-tight">
+          {{ locale === 'nl' ? 'Archief' : 'Archive' }}
```
Comment (ddndauw) [inline frontend/app/components/home/Hero.vue]: Please put this in the json files, thank you.

Comment (Robindr145) [reply frontend/app/components/home/Hero.vue]: will be fixed in next commits

**Review by Alikznollet — APPROVED** [review]
This looks ready to be merged!

**Review by Nowak-Dominik — APPROVED** [review]
You did a good job. Thanks for doing the changes I asked for.
This pr is ready to be merged from my side
